cni/pkg/ambient/net.go

```diff
@@ -50,24 +50,37 @@ func RouteExists(rte []string) bool {
 	return output == "1"
 }
 
-func AddPodToMesh(client kubernetes.Interface, pod *corev1.Pod, ip string) {
-	addPodToMeshWithIptables(pod, ip)
+// AddPodToMesh will actually add a pod IP to the mesh - and will be called once for each IP.
+// In the normal case ( CNI chain ) the IPs are based on the IPAM allocations of CNI plugins before Istio.
+// Normally Istio should be the last in the chain - but not clear we can guarantee this. The Pod may add
```

Comment:
```suggestion
// Normally Istio should be the last in the chain - but not clear we can guarantee this.
```

Yeah I noticed that we assume this - we really shouldn't. Our plugin's order in the chain shouldn't be relevant.

Reply:
I'm stating how it works - not how it should work. I completely agree with you on how it should be.